### DISCUSSION TITLE
Count of hits

### DISCUSSION THREAD

**Discussion by ngms17** [jertel/elastalert2#903] [Q&A] [answered]
Is it possible to have the individual count of hits of a certain rule? This is the case:

I am using the a rule type "any", that runs every minute. Every query hit should give me an alert, which is working as expected. The rule is querying "alert.severity = 1" from Suricata events ingested into elasticsearch.

Lets imagine that from 23:26 to 23:27, elastalert gives me 2 query hits. I can use the "num_hits" field to indicate me the number of hits that elastalert detected.

But, one of those alerts generated more hits one elasticsearch:

![rule](https://user-images.githubusercontent.com/19928515/178529707-84ccbe0b-7d83-4f10-9c87-a0eeb95a6467.PNG)

For Elastalert, this will always be 1 hit, no matter the amount of "hits" i have for that rule on elasticsearch. What can i do to print on my email alert the count (total) of hits for that rule, instead of the hits that elastalert query detected.

I know i can use "realert: 0" to trigger every single alert, but in some cases a rule can have a large number of hits in 1 minute making elastalert generate a lot of alerts.

**Comment by BuffaloWill** [comment, marked as answer]
Have you looked at the aggregation feature?
https://elastalert2.readthedocs.io/en/latest/ruletypes.html?highlight=aggregate#aggregation
